File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        double  rev_num=0;
        int pop=0;
        while(x!=0)
        {
        if (rev_num > INT_MAX/10 || (rev_num == INT_MAX / 10 && pop > 7))
               return 0;
        if (rev_num < INT_MIN/10 || (rev_num == INT_MIN / 10 && pop < -8))
            return 0;
           
            rev_num=rev_num*10+x%10;
            x=x/10;
        }
        return rev_num;
        
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int temp=x;
        int rev=0;
        while(temp!=0){
            int digit=temp%10;
            if (rev > Integer.MAX_VALUE / 10 || rev < Integer.MIN_VALUE / 10) {
                return 0; // Return 0 if overflow occurs
            }
            rev=rev*10+digit;
            temp/=10;
            
        }
        return rev;
    }
}